content/en/blog/_posts/2024-04-29-structured-authentication-configuration-beta.md

```diff
@@ -0,0 +1,216 @@
+---
+layout: blog
+title: "Kubernetes 1.30: Structured Authentication Configuration Moves to Beta"
+date: 2024-04-29
+slug: structured-authentication-moves-to-beta
+---
+
+**Authors:** [Anish Ramasekar](https://github.com/aramase) (Microsoft)
+
+With Kubernetes 1.30, we (SIG Auth) are moving Structured Authentication Configuration to beta.
+
+## Motivation
+Kubernetes has had a long-standing need for a more flexible and extensible
+authentication system. The current system, while powerful, has some limitations
+that make it difficult to use in certain scenarios. For example, it is not
+possible to use multiple authenticators of the same type (e.g., multiple JWT
+authenticators) or to change the configuration without restarting the API server. The
+Structured Authentication Configuration feature is the first step towards
+addressing these limitations and providing a more flexible and extensible way
+to configure authentication in Kubernetes.
+
+## What is Structured Authentication Configuration?
+
+The Structured Authentication Configuration feature is a new way to configure
+authentication in Kubernetes. It currently only supports configuring JWT
+authenticators, which serve as the next iteration of the existing OIDC
+authenticator. JWT authenticator is an authenticator to
+authenticate Kubernetes users using JWT compliant tokens. The authenticator
+will attempt to parse a raw ID token, verify it's been signed by the configured 
+issuer.
+
+The feature is designed to be more flexible and extensible than the existing
+flag-based approach for configuring the JWT authenticator.
+
+### Benefits of Structured Authentication Configuration
+The Structured Authentication Configuration feature brings several benefits to
+Kubernetes:
+1. **Multiple JWT authenticators**: You can configure multiple JWT authenticators
+   simultaneously. This allows you to use multiple identity providers (e.g.,
+   Okta, Keycloak, GitLab) without needing to use an intermediary like Dex
+   that handles multiplexing between multiple identity providers.
+2. **Dynamic configuration**: You can change the configuration without
+   restarting the API server. This allows you to add, remove, or modify
+   authenticators without disrupting the API server.
+3. **Any JWT-compliant token**: You can use any JWT-compliant token for
+   authentication. This allows you to use tokens from any identity provider that
+   supports JWT. The minimum valid JWT payload must contain the claims documented 
+   [here](https://github.com/kubernetes/kubernetes/blob/121607e80963370c1838f9f620c2b8552041abfc/staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1beta1/types.go#L152-L157).
```

Comment:
I've already added the minimum required JWT payload in the Kubernetes doc for v1.30.

Reply:
OK, so: please link to the new doc (you can assume that the _dev-1.30_ branch has merged into _main_ before this article publishes, rather than to the source code.

Our readers often aren't Go coders.